theme/js/components/vanilla/update-url.js

```diff
@@ -0,0 +1,38 @@
+/** @type {Map<string, HTMLElement>} */
+const targets = new Map();
+
+function moveToHash() {
+  console.log("move to hash");
+  const hash = window.location.hash;
+  if(!hash) {
+    return;
+  }
+  let target = hash.slice(1);
+  if(target.includes("/")) {
+    target = target.split("/").shift() || "";
+  }
+  let button;
+  if(button = targets.get(target)) {
+    button.click();
+  }
+}
+
+export default function handleUpdateUrlButtons() {
+  /** @type {NodeListOf<HTMLElement>} */
+  const buttons = document.querySelectorAll("[data-update-url]");
+
+  buttons.forEach((button) => {
+    const target = button.dataset.updateUrl || button.getAttribute("aria-controls");
```

Comment:
This is used to know what hash to add to the URL. 
We can't use `aria-controls` because this cause a glitch on loading : 
- the browser is moving to an anchor that match a tabpanel
- but the tabpanel are translated to the right and hidden with CSS
- and the "current" tabpanel is moved back to view and shown

This causes the browser to move to the bottom of the panel and not the top, which is the expected scroll.

To prevent this, we use the tab button as anchor and pass their id in `data-update-url`.